src/schema/users.ts

```diff
@@ -567,6 +649,91 @@ export const resolvers: IResolvers<any, Context> = {
         )}.png?r=${randomStr}`,
       };
     },
+    updateUserProfile: async (
+      _,
+      { data, upload }: GQLUserParameters,
+      ctx,
+    ): Promise<GQLUser> => {
+      const repo = ctx.con.getRepository(User);
+      const user = await repo.findOne({ id: ctx.userId });
+
+      if (!user) {
+        throw new AuthenticationError(
+          JSON.stringify({ auth: 'User not found' }),
+        );
+      }
+
+      const isUsernameValid = new RegExp(/^@?(\w){1,15}$/).test(data.username);
```

Comment:
I extended it to 39 chars as well in my previous commit

```suggestion
      const isUsernameValid = new RegExp(/^@?(\w){1,39}$/).test(data.username);
```